packages/common-all/src/engine/EngineV3Base.ts

```diff
@@ -291,37 +290,49 @@ export abstract class EngineV3Base implements ReducedDEngine {
     if (vault?.selfContained === "true" || vault?.selfContained === "false")
       vault.selfContained = vault.selfContained === "true";
 
-    const items = await this.fuseEngine.queryNote({
-      qs,
-      onlyDirectChildren,
-      originalQS,
-    });
+    const response = await this.noteQueryEngine.query(qs);
 
-    if (items.length === 0) {
-      return { data: [] };
-    }
+    response
+      .map(async (items) => {
+        if (items.length === 0) {
+          return { data: [] };
+        }
 
-    const notes = await Promise.all(
-      items.map((ent) => this.noteStore.get(ent.id)) // TODO: Should be using metadata instead
-    );
+        const notes = await Promise.all(
+          items.map((ent) => this.noteStore.get(ent.id)) // TODO: Should be using metadata instead
```

Comment:
Reading this I was wondering what if `noteStore.get` throws or returns and `RespV3` with an error.
Read thw whole thread https://github.com/supermacro/neverthrow/issues/113 (its not long) for a discussion around that.